src/App.tsx

```diff
@@ -38,30 +41,42 @@ function App() {
     getJoke();
   }, [url]);
 
-  const numberOfJokesError = () => {
-    if (Number(numberOfJokes) > 100 || Number(numberOfJokes) < 1) return true;
-    return false;
-  };
+  type Values = {
+    counter: string;
+  }
 
-  const downloadJokes = () => {
-    fetch(`${baseUrl}/random/${numberOfJokes}`)
-      .then((response) => {
-        setDownloadError(null);
-        if (!response.ok) throw new Error('Data not fetch.');
-        return response.json();
-      }).then((data) => {
-        const dataToFile = data.value.reduce((cos: string, jokeInLoop:jokeType) => `${cos}${jokeInLoop.joke}\n`, '');
-        const element = document.createElement('a');
-        element.setAttribute('href', `data:text/plain;charset=utf-8, ${encodeURIComponent(dataToFile)}`);
-        element.setAttribute('download', 'jokes.txt');
-        document.body.appendChild(element); // downloadJokes(numberOfJokes);
-        element.click();
-      }).catch((e) => {
-        setDownloadError(e);
-      });
-    return error;
+  const initialValues: Values = {
+    counter: '1',
   };
 
+  const formikDownloadForm = useFormik({
+    initialValues,
+    onSubmit: (value: Values, actions) => {
+      fetch(`${baseUrl}/random/${value.counter}`)
+        .then((response) => {
+          setDownloadError(null);
+          if (!response.ok) throw new Error('Data not fetch.');
+          return response.json();
+        }).then((data) => {
+          const dataToFile = data.value.reduce((cos: string, jokeInLoop:jokeType) => `${cos}${jokeInLoop.joke}\n`, '');
+          const element = document.createElement('a');
+          element.setAttribute('href', `data:text/plain;charset=utf-8, ${encodeURIComponent(dataToFile)}`);
+          element.setAttribute('download', 'jokes.txt');
+          document.body.appendChild(element); // downloadJokes(numberOfJokes);
+          element.click();
+        }).catch((e) => {
+          setDownloadError(e);
+        });
+      actions.setSubmitting(false);
+    },
+    validationSchema: Yup.object({
+      counter: Yup.number()
+        .min(1, 'Too Short!')
```

Comment:
use translations in the error messages, you can create new group of keys 'validationErrors' in en.json